balloon.go

```diff
@@ -0,0 +1,78 @@
+package firecracker
+
+import(
+	models "github.com/firecracker-microvm/firecracker-go-sdk/client/models"
+)
+
+// BalloonBuilder is a builder that will create a balloon used to set up
+// the firecracker microVM.
+type BalloonBuilder struct {
+	balloon models.Balloon
+}
+
+// NewBalloonBuilder will return a new BalloonBuilder with a given amountMB and DeflateOnOom.
+func NewBalloonBuilder(AmountMb int64, DeflateOnOom bool) BalloonBuilder{
+	return BalloonBuilder{}.createBalloon(AmountMb, DeflateOnOom)
+}
+
+// BalloonOpt represents an optional function used to allow for specific
+// customization of the models.Balloon structure.
+type BalloonOpt func(*models.Balloon)
+
+// getBalloon return the balloon
+func (b BalloonBuilder) getBalloon() models.Balloon {
+	return b.balloon
+}
+
+// createBalloon will set the given balloon with new amountMB and DeflateOnOom. 
+// The StatsPollingIntervals of host balloon will be set to 0 by default.
+func (b BalloonBuilder) createBalloon(AmountMb int64, DeflateOnOom bool, opts ...BalloonOpt) BalloonBuilder {
+	b.balloon = models.Balloon {
+		AmountMb: &AmountMb,
+		DeflateOnOom: &DeflateOnOom,
+		StatsPollingIntervals: 0,
+	}
+
+	for _, opt := range opts {
+		opt(&b.balloon)
+	}
+
+	return b
+}
+
+// updateBalloon will set the given balloon with new amountMB and DeflateOnOom. 
+// The StatsPollingIntervals of host balloon will be set to 0 by default.
+func (b BalloonBuilder) updateBalloon(AmountMb int64, opts ...BalloonOpt) BalloonBuilder {
+	b.balloon = models.Balloon {
+		AmountMb: &AmountMb,
+	}
+
+	for _, opt := range opts {
+		opt(&b.balloon)
+	}
+
+	return b
+}
+
+// WithAmountMB sets the target size of the balloon
+func WithAmountMB(amountMB int64) BalloonOpt {
```

Comment:
Is it MB or MiB?

Reply:
I am pretty sure it is in MB, as [this yaml file](https://github.com/firecracker-microvm/firecracker/blob/9e94be71f4a079bf8fafd8f3c801750e744d930f/src/api_server/swagger/firecracker.yaml#L625) is shown.